### PR TITLE
Revises 'mc diff' command help menu

### LINKED ISSUE #2382: mc diff command help menu needs to be revised
## mc version
- (paste output of `mc version`)

### PR DESCRIPTION
**Pull request by ebozduman**
Fixes #2382

### CODE REVIEW

File: cmd/diff-main.go [hunk 1]
```diff
@@ -35,7 +35,7 @@ var (
 // Compute differences between two files or folders.
 var diffCmd = cli.Command{
 	Name:   "diff",
-	Usage:  "List objects with size difference or missing between two folders or buckets.",
+	Usage:  "List differences in object name, size, and date between folders.",
```
Comment (vadmeste) [inline cmd/diff-main.go]: I think 'two folders' will still be more clear.

Comment (ebozduman) [reply cmd/diff-main.go]: @vadmeste,
`between` in English is inherently involves 2 objects/things, unlike `among`, which is used when many or more than 2 items/units are involved in the action. That's why, there is no need to say here `two folders` explicitly.

Comment (vadmeste) [reply cmd/diff-main.go]: > @vadmeste,
> between in English is inherently involves 2 objects/things, unlike among, which is used when many or more than 2 items/units are involved in the action. That's why, there is no need to say here two folders explicitly.

:+1:

File: cmd/diff-main.go [hunk 2]
```diff
@@ -49,21 +49,20 @@ FLAGS:
...
+  compare the differences in object content. 
 
-  Differences between source and destination are represented by mark notations with following meaning:
-    > - indicates file should be copied.
-    < - indicates file should be deleted.
-    ! - indicates file differs in size or type.
+LEGEND:
+    > - object is only in source.
+    < - object is only in destination.
+    ! - object is newer in source.
```
Comment (nitisht) [inline cmd/diff-main.go]: > object is newer in source.

how about `newer object is in source.`

File: cmd/diff-main.go [hunk 3]
```diff
@@ -49,21 +49,20 @@ FLAGS:
   {{range .VisibleFlags}}{{.}}
   {{end}}
 DESCRIPTION:
-  Diff only lists missing objects or objects with size differences. It *DOES NOT* compare contents. 
-  i.e. objects of same name and size, but differing in content are not noticed.
+  Diff only calculates differences in object name, size and time. It *DOES NOT*
+  compare the differences in object content.
```
Comment (nitisht) [inline cmd/diff-main.go]: > Diff only calculates differences in object name, size and time. It *DOES NOT* compare the differences in object content.

How about `Diff only calculates differences in object name, size and last modified time. It *DOES NOT* compare  object content.`

Comment (ebozduman) [reply cmd/diff-main.go]: I've also suggested the same, but @abperiasamy thinks `time` is good enough. 
I'll make the other change (`It *DOES NOT* compare object content.`).

File: cmd/diff-main.go (hunk 2 above)
Comment (ebozduman) [reply cmd/diff-main.go]: OK.

**Pull request merged**